recipes/nuraft/all/patches/1001-cmake_patches.patch

```diff
@@ -0,0 +1,151 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 55d29a8..3af27be 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -1,5 +1,6 @@
+ cmake_minimum_required(VERSION 3.5)
+ project(NuRaft VERSION 1.0.0 LANGUAGES CXX)
++set (CMAKE_CXX_STANDARD 11)
+ 
+ # === Build type (default: RelWithDebInfo, O2) ===========
+ if (NOT CMAKE_BUILD_TYPE)
+@@ -26,41 +27,23 @@ endif()
+ 
+ 
+ # === Find ASIO ===
+-if (BOOST_INCLUDE_PATH AND BOOST_LIBRARY_PATH)
++find_package(OpenSSL CONFIG REQUIRED)
++find_package(Boost CONFIG)
++if (Boost_FOUND)
+     # If Boost path (both include and library) is given,
+     # use Boost's ASIO.
+-    message(STATUS "Boost include path: " ${BOOST_INCLUDE_PATH})
+-    message(STATUS "Boost library path: " ${BOOST_LIBRARY_PATH})
+-
+     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DUSE_BOOST_ASIO")
+-
+-    set(ASIO_INCLUDE_DIR ${BOOST_INCLUDE_PATH})
+-    set(LIBBOOST_SYSTEM "${BOOST_LIBRARY_PATH}/libboost_system.a")
+-
++    set(ASIO_DEP boost::boost)
+ else ()
+     # If not, ASIO standalone mode.
+-    FIND_PATH(ASIO_INCLUDE_DIR
+-              NAME asio.hpp
+-              HINTS ${PROJECT_SOURCE_DIR}/asio/asio/include
+-                    $ENV{HOME}/local/include
+-                    /opt/local/include
+-                    /usr/local/include
+-                    /usr/include)
+-
++    find_package(Asio CONFIG REQUIRED)
+     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DASIO_STANDALONE")
+-
+-endif ()
+-
+-if (NOT ASIO_INCLUDE_DIR)
+-    message(FATAL_ERROR "Can't find ASIO header files")
+-else ()
+-    message(STATUS "ASIO include path: " ${ASIO_INCLUDE_DIR})
++    set(ASIO_DEP asio::asio)
+ endif ()
+ 
+ 
+ # === Includes ===
+ include_directories(BEFORE ./)
+-include_directories(BEFORE ${ASIO_INCLUDE_DIR})
+ include_directories(BEFORE ${PROJECT_SOURCE_DIR}/include)
+ include_directories(BEFORE ${PROJECT_SOURCE_DIR}/include/libnuraft)
+ include_directories(BEFORE ${PROJECT_SOURCE_DIR}/examples)
+@@ -81,24 +64,11 @@ endif()
+ if (NOT WIN32)
+     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall")
+     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-pessimizing-move")
+-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
+-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -g")
+     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fPIC")
+-    if (APPLE)
+-        include_directories(BEFORE
+-                /usr/local/opt/openssl/include
+-                )
+-        link_directories(
+-                /usr/local/opt/openssl/lib
+-                )
+-    else()
+-        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -pthread")
```

Comment:
Okay, it's getting a ride from Boost package which needs pthread, so it will be linked to pthread anyway.